### ISSUE TITLE
Android crash during Achievement load

### ISSUE THREAD

**Issue by gasior** [playgameservices/play-games-plugin-for-unity#1697]
Hi.
Unity 5.5.2f1, Google Play Games 0.9.37, newest play services (10.2.3 - we're part of EAP for Firebase).
We got a crash in what looks like achievement loading.

Based on logcat it looks like user was signed in to play games, then it looks like during achievement loading, com.google.android.gms services was killed. This resulted in revoking auth and crash.

Full logcat attached. Here's cleaned up version

First login during start of the game:
```
04-13 10:17:23.120 10995 12029 I GamesNativeSDK: Auth operation started: SIGN IN
04-13 10:17:23.120 10995 12029 I GamesNativeSDK: Connecting to Google Play...
04-13 10:17:23.175 21341 22588 W GamesServiceBroker: Client connected with SDK 10298000, Services 10298440, and Games 39080040
04-13 10:17:23.388 10995 10995 V GamesNativeSDK: Play Games callback indicates connection failure.
04-13 10:17:23.388 10995 12029 I GamesNativeSDK: UI interaction required to connect to Google Play.
04-13 10:17:23.411 10995 12029 I GamesNativeSDK: Auth operation started: SIGN IN
04-13 10:17:23.412 10995 12029 I GamesNativeSDK: Connecting to Google Play...
04-13 10:17:23.528 10995 10995 V GamesNativeSDK: Play Games callback indicates connection failure.
04-13 10:17:23.528 10995 12029 I GamesNativeSDK: UI interaction required to connect to Google Play.
04-13 10:17:23.528 10995 12029 I GamesNativeSDK: Resolving connection.
04-13 10:17:41.373 10995 10995 V GamesNativeSDK: Play Games callback indicates connection.
04-13 10:17:41.401 10995 12029 I GamesNativeSDK: Successfully connected to Google Play.
04-13 10:17:46.495 10995 12029 V GamesNativeSDK: Detaching from JVM on thread main_dispatch

```

then there's crash in gms service:
```
04-13 10:18:43.451  3723  4208 W ActivityManager: Scheduling restart of crashed service com.google.android.gms/com.google.firebase.auth.api.gms.service.FirebaseAuthService in 1000ms
04-13 10:18:43.451  3723  4208 W ActivityManager: Scheduling restart of crashed service com.google.android.gms/.mdm.services.NetworkQualityAndroidService in 11000ms
04-13 10:18:43.452  3723  4208 W ActivityManager: Scheduling restart of crashed service com.google.android.gms/.measurement.service.MeasurementBrokerService in 21000ms
04-13 10:18:43.452  3723  4208 W ActivityManager: Scheduling restart of crashed service com.google.android.gms/.games.chimera.GamesAndroidServiceProxy in 31000ms
04-13 10:18:43.452  3723  4208 W ActivityManager: Scheduling restart of crashed service com.google.android.gms/.cast.service.CastDeviceScannerIntentService in 118646ms
04-13 10:18:43.453  3723  4208 W ActivityManager: Scheduling restart of crashed service com.google.android.gms/.chimera.container.FileApkIntentOperation$ExternalFileApkService in 40999ms
04-13 10:18:43.453  3723  3981 E ConnectivityService: RemoteException caught trying to send a callback msg for NetworkRequest [ id=22, legacyType=-1, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED] ]
04-13 10:18:43.453  3723  4208 W ActivityManager: Scheduling restart of crashed service com.google.android.gms/com.google.android.location.places.service.GeoDataService in 40999ms
04-13 10:18:43.453  3723  4208 W ActivityManager: Scheduling restart of crashed service com.google.android.gms/.cast.media.CastMediaRouteProviderService in 50998ms
04-13 10:18:43.454  3723  4208 W ActivityManager: Scheduling restart of crashed service com.google.android.gms/.checkin.CheckinService in 60998ms
04-13 10:18:43.454  3723  4208 W ActivityManager: Scheduling restart of crashed service com.google.android.gms/.chimera.GmsIntentOperationService in 128645ms
04-13 10:18:43.454  3723  4208 W ActivityManager: Scheduling restart of crashed service com.google.android.gms/.ads.identifier.service.AdvertisingIdService in 70998ms
04-13 10:18:43.454  3723  4208 W ActivityManager: Scheduling restart of crashed service com.google.android.gms/.games.chimera.GamesAsyncServiceProxy in 80998ms
04-13 10:18:43.455  3723  4208 W ActivityManager: Scheduling restart of crashed service com.google.android.gms/.games.chimera.GamesSyncServiceMainProxy in 80998ms
04-13 10:18:43.455  3723  4208 W ActivityManager: Scheduling restart of crashed service com.google.android.gms/.people.service.PeopleService in 90998ms
04-13 10:18:43.455  3723  4208 W ActivityManager: Scheduling restart of crashed service com.google.android.gms/.appinvite.service.AppInviteService in 100998ms
```

Which happend during play games loading it's data:

```
04-13 10:18:43.589 10995 10995 V GamesNativeSDK: Play Games callback indicates disconnection.
04-13 10:18:43.589 10995 12055 W GamesNativeSDK: Could not automatically seed achievement cache.
04-13 10:18:43.589 10995 12055 W GamesNativeSDK: Could not automatically seed event cache.
04-13 10:18:43.598 14019 14019 D TimaKeyStoreProvider: TimaSignature is unavailable
04-13 10:18:43.598 14019 14019 D ActivityThread: Added TimaKeyStore provider
04-13 10:18:43.610  3723 14464 I ActivityManager: DSS on for com.google.android.googlequicksearchbox and scale is 1.0
04-13 10:18:43.632 10995 12055 I Unity   : Calling Signout in token client
04-13 10:18:43.632 10995 12055 I Unity   :
04-13 10:18:43.632 10995 12055 I Unity   : (Filename: ./artifacts/generated/common/runtime/DebugBindings.gen.cpp Line: 51)
04-13 10:18:43.632 10995 12055 I Unity   :
04-13 10:18:43.633 10995 12055 E art     : JNI ERROR (app bug): accessed stale local reference 0x200001 (index 0 in a table of size 0)
04-13 10:18:43.633 10995 12055 F art     : art/runtime/indirect_reference_table.cc:68] JNI ERROR (app bug): see above.
04-13 10:18:43.640 10995 11128 W Unity   : *** [Play Games Plugin DLL] 04/13/17 10:18:43 +02:00 ERROR: Error retrieving achievements - check the log for more information. Failing signin.
04-13 10:18:43.640 10995 11128 W Unity   :
04-13 10:18:43.640 10995 11128 W Unity   : (Filename: ./artifacts/generated/common/runtime/DebugBindings.gen.cpp Line: 51)
04-13 10:18:43.640 10995 11128 W Unity   :
```

And resulted in crash:
```
04-13 10:18:44.061 10995 12055 E CRASH   : signal 6 (SIGABRT), code -6 (?), fault addr --------
04-13 10:18:44.061 10995 12055 E CRASH   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
04-13 10:18:44.061 10995 12055 E CRASH   : Build fingerprint: 'samsung/hero2ltexx/hero2lte:7.0/NRD90M/G935FXXU1DQB3:user/release-keys'
04-13 10:18:44.061 10995 12055 E CRASH   : Revision: '9'
04-13 10:18:44.061 10995 12055 E CRASH   : pid: 10995, tid: 12055, name: Thread-184  >>> mobi.byss.gun3 <<<
04-13 10:18:44.061 10995 12055 E CRASH   :     r0 00000000  r1 00002f17  r2 00000006  r3 00000008
04-13 10:18:44.061 10995 12055 E CRASH   :     r4 acd93978  r5 00000006  r6 acd93920  r7 0000010c
04-13 10:18:44.061 10995 12055 E CRASH   :     r8 acd93238  r9 ea62e93f  sl acd93218  fp 00000000
04-13 10:18:44.061 10995 12055 E CRASH   :     ip 0000000b  sp acd93148  lr ec0db8c7  pc ec0de130  cpsr 472f4d30
04-13 10:18:44.061 10995 12055 E CRASH   :
04-13 10:18:44.061 10995 12055 E CRASH   : backtrace:
```

Full backtrace attached
[games_crash.txt](https://github.com/playgameservices/play-games-plugin-for-unity/files/921967/games_crash.txt)

I can see on Android samples, that before using any Play Games methods like achievements incrementation etc, there's a mClient.isConnected() check.

Any way we can such check on Unity?

**Comment by Honikou** [comment]
How do you load your board ?

**Comment by claywilkinson** [comment]
Please see #1675 - I believe this is the same issue?

**Comment by claywilkinson** [comment]
I believe this was fixed along with #1675.
